File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.java
class Solution {
    public int minChanges(String s) {
        char currChar = s.charAt(0);
        int cnt = 0, ans = 0;
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == currChar){
                cnt++;
                continue;
            }
            if(cnt%2 == 0) cnt = 1;
            else{
                cnt = 0;
                ans++;
            }
            currChar = s.charAt(i);
        }
        return ans;
    }
}

File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.cpp
class Solution {
public:
    int minChanges(string s) {
        int cnt = 0;
        // Check pairs of characters (i, i+1) with step size 2
        for(int i=0; i<s.size(); i+=2){
            // If characters in current pair don't match, we need one change to make them equal
            if(s[i] != s[i+1]) cnt++;
        }
        return cnt;
    }
};